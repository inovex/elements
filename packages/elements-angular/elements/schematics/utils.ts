import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
