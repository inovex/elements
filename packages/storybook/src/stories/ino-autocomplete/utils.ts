import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
