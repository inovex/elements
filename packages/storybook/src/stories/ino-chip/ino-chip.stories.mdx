import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-chip/readme.md";

<Meta title="Buttons/ino-chip/Documentation" />

<Description markdown={readme} />
