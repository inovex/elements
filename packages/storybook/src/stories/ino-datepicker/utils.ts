import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
