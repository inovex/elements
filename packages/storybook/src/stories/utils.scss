/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
