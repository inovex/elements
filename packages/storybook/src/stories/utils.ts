import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
