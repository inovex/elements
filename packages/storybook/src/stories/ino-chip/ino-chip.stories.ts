import { Components } from '@inovex.de/elements';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator, withColorScheme, withIconControl } from '../utils';

import './ino-chip.scss';

export default {
  title: 'Buttons/<ino-chip>',
  component: 'ino-chip',
  decorators: [defaultDecorator],
  parameters: {
    actions: {
      handles: ['removeChip'],
    },
  },
};

export const Playground: Story<Components.InoChip> = (args) => html`
  <ino-chip
    class="customizable-chip"
    color-scheme="${args.colorScheme}"
    fill="${args.fill}"
    label="${args.label}"
    removable="${args.removable}"
    selectable="${args.selectable}"
    selected="${args.selected}"
    value="${args.value}"
  </ino-chip>
`;

Playground.args = {
  colorScheme: 'primary',
  fill: 'solid',
  label: 'Label',
  removable: false,
  selectable: false,
  selected: false,
  value: '',
};

withColorScheme(Playground, 'colorScheme', 'primary');

Playground.argTypes.fill = {
  control: {
    type: 'select',
  },
  options: ['solid', 'outline'],
};

export const ColorsAndFill = () => html`
  <ino-chip-set>
    <ino-chip label="Default"></ino-chip>
    <ino-chip color-scheme="primary" label="Primary"></ino-chip>
    <ino-chip color-scheme="secondary" label="Secondary"></ino-chip>
    <ino-chip color-scheme="success" label="Success"></ino-chip>
    <ino-chip color-scheme="warning" label="Warning"></ino-chip>
    <ino-chip color-scheme="error" label="Error"></ino-chip>
    <ino-chip color-scheme="light" label="Light"></ino-chip>
    <ino-chip color-scheme="dark" label="Dark"></ino-chip>
  </ino-chip-set>
  <ino-chip-set>
    <ino-chip fill="outline" label="Outline"></ino-chip>
    <ino-chip fill="outline" label="Primary" color-scheme="primary"></ino-chip>
    <ino-chip
      fill="outline"
      label="Secondary"
      color-scheme="secondary"
    ></ino-chip>
    <ino-chip fill="outline" color-scheme="success" label="Success"></ino-chip>
    <ino-chip fill="outline" color-scheme="warning" label="Warning"></ino-chip>
    <ino-chip fill="outline" color-scheme="error" label="Error"></ino-chip>
    <ino-chip fill="outline" color-scheme="light" label="Light"></ino-chip>
    <ino-chip fill="outline" color-scheme="dark" label="Dark"></ino-chip>
  </ino-chip-set>
`;

export const Icons = () => html`
  <ino-chip-set>
    <ino-chip color-scheme="primary" label="Leading">
      <ino-icon slot="icon-leading" icon="checkmark"></ino-icon>
    </ino-chip>
    <ino-chip color-scheme="primary" label="Trailing">
      <ino-icon slot="icon-trailing" icon="checkmark"></ino-icon>
    </ino-chip>
    <ino-chip color-scheme="primary" label="Leading & Trailing">
      <ino-icon slot="icon-leading" icon="checkmark"></ino-icon>
      <ino-icon slot="icon-trailing" icon="checkmark"></ino-icon>
    </ino-chip>
  </ino-chip-set>
`;

export const States = () => html`
  <h4>Removable and states</h4>
  <ino-chip-set>
    <ino-chip color-scheme="primary" removable label="Removable"></ino-chip>
    <ino-chip color-scheme="primary" selected label="Selected"></ino-chip>
    <ino-chip color-scheme="primary" selected label="Selected with Checkmark" selectable></ino-chip>
  </ino-chip-set>
  </div>
`;
