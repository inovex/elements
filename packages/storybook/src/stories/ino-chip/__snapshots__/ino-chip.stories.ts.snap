// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/<ino-chip> Colors smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <div class="ino-chip-story">
      <ino-chip class="ino-chip--color-scheme-default ino-chip--solid hydrated">
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Default
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="primary"
                class="ino-chip--color-scheme-primary ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Primary
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="secondary"
                class="ino-chip--color-scheme-secondary ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Secondary
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="success"
                class="ino-chip--color-scheme-success ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Success
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="warning"
                class="ino-chip--color-scheme-warning ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Warning
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="error"
                class="ino-chip--color-scheme-error ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Error
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="light"
                class="ino-chip--color-scheme-light ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Light
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="dark"
                class="ino-chip--color-scheme-dark ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Dark
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
    </div>
    <div class="ino-chip-story">
      <ino-chip fill="outline"
                class="ino-chip--color-scheme-default ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Outline
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip fill="outline"
                color-scheme="primary"
                class="ino-chip--color-scheme-primary ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Primary
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip fill="outline"
                color-scheme="secondary"
                class="ino-chip--color-scheme-secondary ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Secondary
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip fill="outline"
                color-scheme="success"
                class="ino-chip--color-scheme-success ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Success
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip fill="outline"
                color-scheme="warning"
                class="ino-chip--color-scheme-warning ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Warning
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip fill="outline"
                color-scheme="error"
                class="ino-chip--color-scheme-error ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Error
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip fill="outline"
                color-scheme="light"
                class="ino-chip--color-scheme-light ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Light
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip fill="outline"
                color-scheme="dark"
                class="ino-chip--color-scheme-dark ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Dark
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
    </div>
  </div>
</div>
`;

exports[`Buttons/<ino-chip> Fill smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <div class="ino-chip-story">
      <ino-chip fill="solid"
                class="ino-chip--color-scheme-default ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip Solid
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip fill="outline"
                class="ino-chip--color-scheme-default ino-chip--outline hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip Outline
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
    </div>
  </div>
</div>
`;

exports[`Buttons/<ino-chip> Filter smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <div class="ino-chip-story">
      <ino-chip selectable
                value="Chip 1"
                class="ino-chip--color-scheme-default ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container mdc-evolution-chip--selectable mdc-evolution-chip--deselecting mdc-evolution-chip--filter mdc-evolution-chip--with-primary-graphic"
              role="presentation"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <span class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                  role="option"
                  tabindex="0"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__graphic">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 1
              </span>
            </span>
          </span>
        </span>
      </ino-chip>
      <ino-chip selectable
                value="Chip 2"
                class="ino-chip--color-scheme-default ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container mdc-evolution-chip--selectable mdc-evolution-chip--deselecting mdc-evolution-chip--filter mdc-evolution-chip--with-primary-graphic"
              role="presentation"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <span class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                  role="option"
                  tabindex="0"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__graphic">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 2
              </span>
            </span>
          </span>
        </span>
      </ino-chip>
      <ino-chip selectable
                value="Chip 3"
                class="ino-chip--color-scheme-default ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container mdc-evolution-chip--selectable mdc-evolution-chip--deselecting mdc-evolution-chip--filter mdc-evolution-chip--with-primary-graphic"
              role="presentation"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <span class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                  role="option"
                  tabindex="0"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__graphic">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 3
              </span>
            </span>
          </span>
        </span>
      </ino-chip>
    </div>
  </div>
</div>
`;

exports[`Buttons/<ino-chip> Icons smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <div class="ino-chip-story">
      <ino-chip>
        Leading
        <ino-icon slot="icon-leading"
                  icon="star"
        >
        </ino-icon>
      </ino-chip>
      <ino-chip>
        Trailing
        <ino-icon slot="icon-trailing"
                  icon="star"
        >
        </ino-icon>
      </ino-chip>
      <ino-chip>
        <span>
          Leading &amp; Trailing
        </span>
        <ino-icon slot="icon-leading"
                  icon="star"
        >
        </ino-icon>
        <ino-icon slot="icon-trailing"
                  icon="star"
        >
        </ino-icon>
      </ino-chip>
    </div>
  </div>
</div>
`;

exports[`Buttons/<ino-chip> Playground smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-chip class="customizable-chip ino-chip--color-scheme-default ino-chip--solid hydrated"
              color-scheme="default"
              disabled="false"
              fill="solid"
              removable="false"
              selectable="false"
              selected="false"
              value
    >
      <span class="mdc-evolution-chip ino-chip-container"
            role="row"
      >
        <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
              role="gridcell"
        >
          <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                  tabindex="0"
                  data-value
                  type="button"
          >
            <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
            </span>
            <span class="mdc-evolution-chip__text-label ino-chip-label">
              Label
            </span>
          </button>
        </span>
      </span>
    </ino-chip>
  </div>
</div>
`;

exports[`Buttons/<ino-chip> Remove smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <div class="ino-chip-story">
      <ino-chip value="Chip 1"
                removable="false"
                class="ino-chip--color-scheme-default ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    data-value="Chip 1"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 1
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip value="Chip 2"
                removable="true"
                class="ino-chip--color-scheme-default ino-chip--solid"
      >
        <span class="mdc-evolution-chip ino-chip-container mdc-evolution-chip--with-trailing-action"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    data-value="Chip 2"
                    data-mdc-deletable
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 2
              </span>
            </button>
          </span>
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--trailing"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--trailing"
                    type="button"
                    tabindex="-1"
                    data-mdc-deletable="true"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--trailing">
              </span>
              <span class="mdc-evolution-chip__icon mdc-evolution-chip__icon--trailing ino-chip-trailing-icon">
                <ino-icon class="ino-chip-close-icon"
                          tabindex="0"
                          role="button"
                >
                </ino-icon>
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip value="Chip 3"
                removable="true"
                class="ino-chip--color-scheme-default ino-chip--solid"
      >
        <span class="mdc-evolution-chip ino-chip-container mdc-evolution-chip--with-trailing-action"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    data-value="Chip 3"
                    data-mdc-deletable
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 3
              </span>
            </button>
          </span>
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--trailing"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--trailing"
                    type="button"
                    tabindex="-1"
                    data-mdc-deletable="true"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--trailing">
              </span>
              <span class="mdc-evolution-chip__icon mdc-evolution-chip__icon--trailing ino-chip-trailing-icon">
                <ino-icon class="ino-chip-close-icon"
                          tabindex="0"
                          role="button"
                >
                </ino-icon>
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
    </div>
  </div>
</div>
`;

exports[`Buttons/<ino-chip> Selection smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <div id="chip-selection"
         class="ino-chip-story"
    >
      <ino-chip color-scheme="primary"
                value="Chip 1"
                class="ino-chip--color-scheme-primary ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    data-value="Chip 1"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 1
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="default"
                value="Chip 2"
                class="ino-chip--color-scheme-default ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    data-value="Chip 2"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 2
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
      <ino-chip color-scheme="default"
                value="Chip 3"
                class="ino-chip--color-scheme-default ino-chip--solid hydrated"
      >
        <span class="mdc-evolution-chip ino-chip-container"
              role="row"
        >
          <span class="mdc-evolution-chip__cell mdc-evolution-chip__cell--primary"
                role="gridcell"
          >
            <button class="mdc-evolution-chip__action mdc-evolution-chip__action--primary"
                    tabindex="0"
                    data-value="Chip 3"
                    type="button"
            >
              <span class="mdc-evolution-chip__ripple mdc-evolution-chip__ripple--primary">
              </span>
              <span class="mdc-evolution-chip__text-label ino-chip-label">
                Chip 3
              </span>
            </button>
          </span>
        </span>
      </ino-chip>
    </div>
  </div>
</div>
`;
